File: Minimum times A has to be repeated such that B is a substring of it - GFG/minimum-times-a-has-to-be-repeated-such-that-b-is-a-substring-of-it.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out=new PrintWriter(System.out);
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String A = read.readLine();
            String B = read.readLine();

            Solution ob = new Solution();
            out.println(ob.minRepeats(A,B));
        }
        out.close();
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution {
    static int minRepeats(String A, String B) {
        // code here
        int count =1;
        String rep =A;
        while (rep.length()<B.length()){
            rep +=A;
            count++;
            
        }
        if (rep.contains(B))
        return count ;
        
        rep +=A;
        if (rep.contains(B))
        return count+1;
        
        
        return -1;
    }
};

File: Minimum times A has to be repeated such that B is a substring of it - GFG/minimum-times-a-has-to-be-repeated-such-that-b-is-a-substring-of-it.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++

class Solution {
  public:
       int minRepeats(string A, string B) {
        // code here
        if(A==B)return 1;
        int i=0,n=A.length(),m=B.length(),ans=1,j=0;
        while(i<n)
        {
            if(A[i]==B[j]){
                j++;
            }
            else{
                if(A[i]==B[0])j=1;
                else j=0;
            }
            i++;
        }
        i=0;
        if(j!=m)ans++;
        while(j<m and A[i]==B[j]){
            i++;
            i%=n;
            if(i==0)ans++;
            j++;
        }
        if(j!=m)return -1;
        if(i==0)ans--;
        return ans;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    scanf("%d ",&t);
    while (t--) {
        string A,B;
        getline(cin,A);
        getline(cin,B);

        Solution ob;
        cout << ob.minRepeats(A,B) << endl;
    }
    return 0;
}
// } Driver Code Ends